e2e_test/source_inline/vault/vault_secret_ddl.slt

```diff
@@ -0,0 +1,60 @@
+control substitution on
+
+# Test for HashiCorp Vault secret backend
+
+# Test with token auth - using real vault server from docker-compose
+statement ok
+CREATE SECRET vault_token_secret
+WITH (
+  backend = 'hashicorp_vault',
+  addr = 'http://vault-server:8200',
+  path = 'secret/data/myapp/db',
+  field = 'password',
+  auth_method = 'token',
+  auth_token = 'root-token',
+  tls_skip_verify = 'true'
+) AS NULL;
+
+query T
+SHOW SECRETS;
+----
+vault_token_secret
+
+
+# Test error case - invalid field
+statement error
+CREATE SECRET vault_invalid_field
+WITH (
+  backend = 'hashicorp_vault',
+  addr = 'http://vault-server:8200',
+  path = 'secret/data/myapp/db',
+  field = 'nonexistent_field',
+  auth_method = 'token',
+  auth_token = 'root-token'
```

Comment:
Can we use `secret` here 🤔 . I suppose we can bootstrap by storing the secret inside RW.